internal/common/teleport.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/cluster-test-suites/internal/helper"
+	"github.com/giantswarm/cluster-test-suites/internal/state"
+	"github.com/giantswarm/clustertest/pkg/logger"
+	"github.com/giantswarm/clustertest/pkg/wait"
+
+	tc "github.com/gravitational/teleport/api/client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func runTeleport(teleportSupported bool) {
+	Context("teleport", func() {
+		var teleportClient *tc.Client
+
+		isClusterRegistered := func(clusterName string) (bool, error) {
+			clusters, err := teleportClient.GetKubernetesServers(context.Background())
+			if err != nil {
+				return false, err
+			}
+			for _, cluster := range clusters {
+				if strings.Contains(cluster.GetName(), clusterName) {
+					logger.Log("cluster registered %v", cluster)
+					return true, nil
+				}
+			}
+			logger.Log("cluster %s still not registered", clusterName)
+			return false, nil
+		}
+
+		BeforeEach(func() {
+			if strings.TrimSpace(os.Getenv("TELEPORT_IDENTITY_FILE")) == "" {
+				Skip("TELEPORT_IDENTITY_FILE env var not set, skipping teleport test")
+				return
+			}
+			var err error
+			teleportClient, err = helper.NewTeleportClient(context.Background(), os.Getenv("TELEPORT_IDENTITY_FILE"))
```

Comment:
I have the same question, I don't see any documentation updates related to this. This isn't set in the CI/CD pipelines.

Reply:
So, it will be set in CI/CD pipeline, [similar to GITHUB_TOKEN_FILE](https://github.com/giantswarm/tekton-resources/blob/1e3521aaa5d09a214b92f81c7add8fdde19015b0/tekton-resources/tasks/run-wc-e2e-tests.yaml#L69-L70)

However, with Teleport we can't really create long lived token, so we need to deploy an tbot agent in the cluster similar to how we are doing with teleport-operator, that periodically refreshes the token.

Reply:
Can we please tackle that before this PR gets merged in? It's no use merging this in if it's not going to actually be used.

Reply:
Yep, won't be merging this until identity file is sorted out.

Reply:
I've added the `do-not-merge/hold` label to be sure :)

Reply:
`teleport-tbot` deployed in `gazelle/cicddev` and identity file sorted.